File: Crio/crio_programming_interview_problems-master/TowerOfHanoi/TowerOfHanoi.cpp
#include <bits/stdc++.h> 
using namespace std; 
//CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
vector<string> movement;
void steps(int n, char from_rod,char to_rod, char aux_rod) 
{ 
	if (n == 1) 
	{ 
		string s=to_string(n)+" "+from_rod+" "+to_rod;
		movement.push_back(s);
		return; 
	} 
	steps(n - 1, from_rod, aux_rod, to_rod); 
	string s=to_string(n)+" "+from_rod+" "+to_rod;
	movement.push_back(s); 
	steps(n - 1, aux_rod, to_rod, from_rod); 
}
//CRIO_SOLUTION_END_MODULE_L1_PROBLEM
vector<string> towerOfHanoi(int n)
{
	//CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
	steps(n,'A','C','B');
	return movement;
	//CRIO_SOLUTION_END_MODULE_L1_PROBLEM
} 

int main() 
{ 
	int n;
	cin>>n;
	vector<string> ans=towerOfHanoi(n);
	for(auto i=0;i<ans.size();i++)
	{
		cout<<ans[i]<<"\n";
	}
} 

File: Crio/crio_programming_interview_problems-master/TowerOfHanoi/TowerOfHanoi.py
#CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
def steps(n, fromRod, toRod, auxRod, result):
    if n == 1:
        result.append("{} {} {}".format(n, fromRod, toRod))
        return
    
    steps(n - 1, fromRod, auxRod, toRod, result)
    result.append("{} {} {}".format(n, fromRod, toRod))
    steps(n - 1, auxRod, toRod, fromRod, result)
#CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
def towerOfHanoi(n):
    #CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    result = []
    steps(n, 'A', 'C', 'B', result)
    return result
    #CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

def main():
    n = int(input())
    result = towerOfHanoi(n)
    for line in result:
        print(line)

if __name__ == '__main__':
    main()
